vault/logical_system.go

```diff
@@ -2943,6 +2943,28 @@ func (b *SystemBackend) pathInternalUIMountRead(ctx context.Context, req *logica
 	return resp, nil
 }
 
+func (b *SystemBackend) pathInternalCountersRequests(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	counters, err := b.Core.loadAllRequestCounters(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	buf, err := json.Marshal(counters.Dated)
+	if err != nil {
+		return nil, err
+	}
+
+	resp := &logical.Response{
+		Data: map[string]interface{}{
+			logical.HTTPStatusCode:  200,
+			logical.HTTPRawBody:     buf,
```

Comment:
Could you walk me through why you're using a raw response here?

Reply:
I don't know any better, and copied from somewhere else.  What should I be doing?

Reply:
We usually return structured data here and rely on Vault's framework to convert it to json. So something like:
```
resp := &logical.Response{
		Data: map[string]interface{
                         "counters": counters.Dated,
                },
}
```

Reply:
Although I thought we were going to return a map with the date range as the key, is that still the case?

Reply:
Well... I noted that this implementation of the HTTP API was a WIP partly because I'm on the fence as to whether to re-negotiate this.  I kind of like this format, it's both more flexible as to time intervals, and it feels more natural to parse full date strings rather than have to hand craft something to translate my proposed month-based nested format.  If you have any thoughts on the matter let me know.

So I'll get back to you once I make up my mind.  For now it's just a way to query the counters from outside the process.

Reply:
I guess I never got to back to you.  In the end I did prefer to keep this response layoutl